tests/db/test_v6_to_v7.py

```diff
@@ -0,0 +1,145 @@
+"""
+    Copyright 2020 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import datetime
+import os
+import uuid
+from dataclasses import dataclass
+from typing import AsyncIterator, Awaitable, Callable, List
+
+import pytest
+from asyncpg import Connection, Record
+from asyncpg.cursor import Cursor
+
+from db.common import PGRestore
+from inmanta import data
+from inmanta.server.services.databaseservice import DatabaseService
+
+
+@pytest.fixture
+async def migrate_v6_to_v7(
+    hard_clean_db, hard_clean_db_post, postgresql_client: Connection, async_finalizer, server_config
+) -> AsyncIterator[Callable[[], Awaitable[None]]]:
+    """
+    Performs a v6 database restore and yields a function that migrates to v7 when called.
+
+    Word of caution:
+    This fixture differs from similar ones in previous version migration tests in the following ways:
+    - It yields a migration function instead of migrating before yielding. This to allow tests to set up desired preconditions.
+    - It only starts the database slice, not the full server. This to make sure agent instance documents after yielding are only
+        affected by the migration, not by any agent management of the server. This prevents agent instance documents expiring
```

Comment:
at restart, all sessions can/should be expired so the migration is allowed to do that as well.

Reply:
My reasoning here was that this makes it difficult to test if the expired state is correctly merged. Though perhaps that doesn't need to be tested then?

Reply:
@wouterdb what's your opinion on this and [the above](https://github.com/inmanta/inmanta/pull/2427#discussion_r508353314)? If the server is expected to expire all sessions at startup, this seems to imply that the merge procedure could be simplified to just ignore the expired state instead of merging them. This would also simplify the required state in the db dump because the expired state merging needs not be tested.

Reply:
I would ignore it, we are talking about a VERY transient thing. (migration of an effect cause by a bug)

Reply:
afa19f9a and 27895b89